examples/contrib/detection/faster_rcnn.py

```diff
@@ -0,0 +1,286 @@
+
+                target_labels = [Dataset.class2name[l.item()] for l in target["labels"]]
+                target_boxes = target["boxes"].long()
+                image = draw_bounding_boxes(image, target_boxes, target_labels, colors="green")
+
+                aim_images.append(aim.Image(image.numpy().transpose((1, 2, 0))))
+        aim_logger.experiment.track(aim_images, name="vis", step=trainer.state.epoch)
+
+    losses = ["loss_classifier", "loss_box_reg", "loss_objectness", "loss_rpn_box_reg", "loss_average"]
+    for loss_name in losses:
+        RunningAverage(output_transform=lambda x: x[loss_name]).attach(trainer, loss_name)
+    ProgressBar().attach(trainer, losses)
+    ProgressBar().attach(evaluator)
+
+    objects_to_checkpoint = {"trainer": trainer, "model": model, "optimizer": optimizer, "lr_scheduler": scheduler}
+    checkpoint = Checkpoint(
+        to_save=objects_to_checkpoint,
+        save_handler=DiskSaver(log_dir, require_empty=False),
+        n_saved=3,
+        score_name="mAP",
+        global_step_transform=lambda *_: trainer.state.epoch,
+    )
+    evaluator.add_event_handler(Events.EPOCH_COMPLETED, checkpoint)
+
+    aim_logger.log_params(
+        {
+            "lr": lr,
+            "image_size": image_size,
+            "batch_size": batch_size,
+            "epochs": epochs,
+        }
+    )
+    aim_logger.attach_output_handler(
+        trainer, event_name=Events.ITERATION_COMPLETED, tag="train", output_transform=lambda loss: loss
+    )
+    aim_logger.attach_output_handler(
+        evaluator,
+        event_name=Events.EPOCH_COMPLETED,
+        tag="val",
+        metric_names=["mAP"],
+        global_step_transform=global_step_from_engine(trainer, Events.ITERATION_COMPLETED),
+    )
+
+    trainer.run(train_dataloader, max_epochs=epochs)
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser()
+    parser.add_argument(
+        "--dataset-root",
+        type=str,
+        help="dataset root directory for VOC2012 Dataset",
+        required=False,
+        default="./dataset",
+    )
+    parser.add_argument("--log-dir", type=str, help="where to put all the logs", required=False, default="./log")
+    parser.add_argument("--epochs", type=int, default=13, help="number of epochs to train (default: 13)")
+    parser.add_argument("--batch-size", type=int, default=4, help="batch size to use")
+    parser.add_argument("--lr", type=float, default=0.01, help="initial lr to use")
+    parser.add_argument(
+        "--download", action="store_true", default=False, help="either to download dataset automatically"
+    )
+    parser.add_argument("--device", type=str, default="cuda", help="gpu or cpu")
+    parser.add_argument("--image-size", type=int, default=512, help="image size to train")
+    parser.add_argument("--experiment-name", type=str, default="test", help="name of one experiment")
+    args = parser.parse_args()
+    run(
```

Comment:
idist could be used for ddp

Reply:
I don't have 2 GPUs to test distribution now. I'll try to write the distribution code and, maybe test on AWS?